### ISSUE TITLE
Add test for each app

### ISSUE THREAD

**Issue by tdowrick** [UCL/scikit-surgeryutils#10]
In GitLab by @ThomasDowrick on Feb 8, 2019, 12:09

Changes to the other SNAPPY libraries could break the demo apps, if any of the functions change, which might not be discovered until someone tries to use it. For example, I just got an error in the sksurgeryrendermodelslikecamera app when I tried to run it (looks like one of the fuctions it uses has been renamed).

We should add a simple test for each app (i.e. does it run OK), and encourage people to implement tests for new apps when they make them.

@MattClarkson @StephenThompson

**Comment by tdowrick** [comment]
In GitLab by @MattClarkson on Feb 8, 2019, 12:24

Agreed.

**Comment by tdowrick** [comment]
In GitLab by @StephenThompson on Feb 8, 2019, 12:42

Is it possible to trigger a test when one of the dependencies is updated?

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 8, 2019, 12:52

I was just thinking about this, looks like we can add some trigger to get it to run automatically:  
https://weisslab.cs.ucl.ac.uk/WEISS/SoftwareRepositories/SNAPPY/scikit-surgeryutils/settings/ci_cd

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 10:07

mentioned in merge request !3

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 10:07

created merge request !3 to address this issue

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 13:25

mentioned in commit 893d5a018aee625df1506a60e191e8876526528a

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 13:25

mentioned in commit 0a2471daee14c03e7063ca0156bbd513f58b3a99

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 13:25

mentioned in commit b3f609c26496d2dd55f8ba706ecf0c1579662f8a

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 13:25

mentioned in commit 987220b5c341fd71a434c56df9341ec0490ea74b

**Comment by tdowrick** [comment]
In GitLab by @ThomasDowrick on Feb 12, 2019, 14:19

I have added some basic tests for each app.

I found that there is a qt plugin for pytest (https://pypi.org/project/pytest-qt/), that I started to use for these tests and have added to requirements-dev.txt.

It seems to make things a bit simpler, as it handles the QApplication creation/deletion automatically, and there is some other functionality that might be useful when we have more GUIs to test.
